torchgeo/datasets/solar_plants_brazil.py

```diff
@@ -0,0 +1,189 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""SolarPlantsBrazil dataset."""
+
+import glob
+import os
+from collections.abc import Callable
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rasterio
+import torch
+from huggingface_hub import snapshot_download
+from matplotlib.figure import Figure
+from torch import Tensor
+
+from torchgeo.datasets import NonGeoDataset
+from torchgeo.datasets.utils import Path
+
+
+class SolarPlantsBrazil(NonGeoDataset):
+    """Solar Plants Brazil dataset (semantic segmentation for photovoltaic detection).
+
+    The `Solar Plants Brazil <https://huggingface.co/datasets/FederCO23/solar-plants-brazil>`__
+    dataset provides satellite imagery and pixel-level annotations for detecting photovoltaic
+    solar power stations.
+
+    Dataset features:
+    * 272 RGB+NIR GeoTIFF images (256x256 pixels)
+    * Binary masks indicating presence of solar panels (1 = panel, 0 = background)
+    * Organized into `train`, `val`, and `test` splits
+    * Float32 GeoTIFF files for both input and mask images
+    * Spatial metadata included (CRS, bounding box), but not used directly for training
+
+    Folder structure:
+    root/train/input/img(123).tif
+    root/train/labels/target(123).tif
+
+    Access:
+    * Dataset is hosted on Hugging Face: https://huggingface.co/datasets/FederCO23/solar-plants-brazil
+    * Code and preprocessing steps available at: https://github.com/FederCO23/UCSD_MLBootcamp_Capstone
+
+    If you use this dataset, please cite or reference the project repository.
+
+    .. versionadded:: 0.6
+    """
+
+    url = 'https://huggingface.co/datasets/FederCO23/solar-plants-brazil'
+    bands = ('Red', 'Green', 'Blue', 'NIR')
```

Comment:
I wonder if we could start using enums - [e.g. like terratorch ](https://github.com/IBM/terratorch/blob/main/terratorch/datasets/utils.py
)

Reply:
I've thought about it, but there's a lot more info I would like to add (min/max/mean wavelength, bandwidth, etc.). I think a formal class would be better. But that's for another day, we would have to update all existing datasets first.